lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,243 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
+
+import '../api/model/model.dart';
+import 'content.dart';
+import 'page.dart';
+import 'clipboard.dart';
+import 'store.dart';
+
+// TODO: Add index of the image preview in the message, to not break if
+//   there are multiple image previews with the same URL in the same
+//   message. Maybe keep `src`, so that on exit the lightbox image doesn't
+//   fly to an image preview with a different URL, following a message edit
+//   while the lightbox was open.
+class _LightboxHeroTag {
+  _LightboxHeroTag({required this.messageId, required this.src});
+
+  final int messageId;
+  final String src;
+
+  @override
+  bool operator ==(Object other) {
+    return other is _LightboxHeroTag &&
+      other.messageId == messageId &&
+      other.src == other.src;
```

Comment:
```suggestion
      other.src == src;
```

:slightly_smiling_face: